### ISSUE TITLE
[Question]: How to embed fabric within another application?

### ISSUE THREAD

**Issue by oskapt** [danielmiessler/fabric#828]
### What is your question?

I use Fabric almost every day, and I sat down last night to build a Python app that incorporates Fabric patterns to perform a series of tasks. Congratulations on moving it over to Go, but it appears that in doing so, Fabric can no longer be incorporated into another app. In the Python version there mention of a `server` that I could never find a way to use, and the docs touted how one could easily load the libraries and use it from within another app. Has all of that been scrapped for a standalone client-only binary, or will there be any sort of API functionality built into the new version, enabling it to run in a central location and field requests from multiple clients?

**Comment by xssdoctor** [comment]
I actually did build a whole api with web frontend but it kind of sucks at the moment and isn't ready for prime time. I'll keep working on it

**Comment by 6thfdwp** [comment]
@xssdoctor  I assume you also use Go to build REST api for this web Frontend to consume? Curious to know what make it sucks?

**Comment by 6thfdwp** [comment]
I was thinking about similar things to further remove friction of interacting AI models without installing anything, CLI etc, it would benefit others without coding background. (though I think it's also pretty cool to pipe a few commands to get insights right in CLI). 
Keen to get involved to help in anyway if we have a plan to roll out some web app. I've been using Go and React a lot in production ready services, quite like Go's pragmatic efficiency. 

Cheers,

**Comment by ksylvan** [comment]
Look at the two web ui applicatiosn that use the REST API from the Go version.

This is an old issue and I'm closing it.

**Comment by oskapt** [comment]
Yeah, this is a super old issue. I stopped using Fabric ages ago.

**Comment by ksylvan** [comment]
Fabric has grown a lot and is extremely useful and usable.

It saves me many hours every single day and is part of my workflow. For example, I just saved myself a few hours by summarizing and grabbing the relevant pieces of two long (2 hour) podcasts and deciding what segments I acually wanted to watch.

I also use it in my development flow every day, using the "summarize_git_diff" and "write_pull-request" patterns several times per day.

I also recently added other generative AI tools (like text-to-speech and image generation) backed by various frontier models.
